File: Back/서지현/piuda/src/main/java/com/example/demo/model/Review.java
package com.example.demo.model;

import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

@Entity
public class Review {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long review_id;
	
	private LocalDate review_date;
	
    private String user_name; // 사용자 이름 필드 추가

    public String getUser_name() {
        return user_name;
    }

    public void setUser_name(String user_name) {
        this.user_name = user_name;
    }

	
	@OneToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "loan_id")
	private Loan loan;

	 public Long getLoan_id() {
	        if (loan != null) {
	            return loan.getLoan_id();
	        }
	        return null; // 대출 정보가 없을 경우 null 반환
	    }
	
    public Long getUserId() {
        if (loan != null && loan.getUser() != null) {
            return loan.getUser().getId();
        }
        return null; // 사용자 정보가 없을 경우 null 반환
    }
    
	private int review_score;
	private String review_content;
	
	public Long getReview_id() {
		return review_id;
	}
	
	public void setReview_id(Long review_id) {
		this.review_id = review_id;
	}
	
	public LocalDate getReview_date() {
		return review_date;
	}
	
	public void setReview_date(LocalDate review_date) {
		this.review_date = review_date;
	}
	
	public Loan getLoan() {
		return loan;
	}
	
	public void setLoan(Loan loan) {
		this.loan = loan;
	}
	
	public int getReview_score() {
		return review_score;
	}
	
	public void setReview_score(int review_score) {
		this.review_score = review_score;
	}
	
	public String getReview_content() {
		return review_content;
	}
	
	public void setReview_content(String review_content) {
		this.review_content = review_content;
	}
}

File: Front/서지현/piuda/lib/review.dart
class Review {
  final String? userName; // 널이 될 수 있음을 표시
  final int? reviewScore; // 널이 될 수 있음을 표시
  final String? reviewContent; // 널이 될 수 있음을 표시
  final String? reviewDate; // 널이 될 수 있음을 표시

  Review({
    this.userName,
    this.reviewScore,
    this.reviewContent,
    this.reviewDate,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      userName: json['user_name'] as String?, // 널을 허용하는 캐스팅
      reviewScore: json['review_score'] as int?, // 널을 허용하는 캐스팅
      reviewContent: json['review_content'] as String?, // 널을 허용하는 캐스팅
      reviewDate: json['review_date'] as String?, // 널을 허용하는 캐스팅
    );
  }
}
